api/v1alpha1/oidc_types.go

```diff
@@ -36,20 +36,21 @@ type OIDC struct {
 	// specified.
 	// +optional
 	Scopes []string `json:"scopes,omitempty"`
+
+	// The redirect URL to be used in the OIDC
+	// [Authentication Request](https://openid.net/specs/openid-connect-core-1_0.html#AuthRequest).
+	// If not specified, uses the default redirect URI "%REQ(x-forwarded-proto)%://%REQ(:authority)%/oauth2/callback"
+	RedirectURL *string `json:"redirectURL,omitempty"`
```

Comment:
this should be `redirect_uri`
Also just noticed,  below fields should be `authorization_uri` and `token_uri`

Reply:
> Also just noticed, below fields should be authorization_uri and token_uri

This might be a bit confusing with the `Endpoint` of a Cluster in Envoy, but I'm using the terminology from the OIDC specification, and Envoy oauth2 filter also uses the same terms.

https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/filters/http/oauth2/v3/oauth.proto#envoy-v3-api-msg-extensions-filters-http-oauth2-v3-oauth2

https://openid.net/specs/openid-connect-core-1_0.html#AuthorizationEndpoint
```
3.1.2.  Authorization Endpoint
The Authorization Endpoint performs Authentication of the End-User. This is done by sending the User Agent to the Authorization Server's Authorization Endpoint for Authentication and Authorization, using request parameters defined by OAuth 2.0 and additional parameters and parameter values defined by OpenID Connect.

Communication with the Authorization Endpoint MUST utilize TLS. See [Section 16.17](https://openid.net/specs/openid-connect-core-1_0.html#TLSRequirements) for more information on using TLS.

https://openid.net/specs/openid-connect-core-1_0.html#TokenEndpoint
```
3.1.3.  Token Endpoint
To obtain an Access Token, an ID Token, and optionally a Refresh Token, the RP (Client) sends a Token Request to the Token Endpoint to obtain a Token Response, as described in Section 3.2 of [OAuth 2.0](https://openid.net/specs/openid-connect-core-1_0.html#RFC6749) [RFC6749], when using the Authorization Code Flow.

Communication with the Token Endpoint MUST utilize TLS. See [Section 16.17](https://openid.net/specs/openid-connect-core-1_0.html#TLSRequirements) for more information on using TLS.
```

Reply:
> this should be `redirect_uri`

Some of the APIs are using `URL`, for example, the rate limit server, and some are using `URI`, should they be unified?

Based on a quick search, it appears `URL` is more appropriate here.

```
Should I use URL or URI?
It depends on what you are trying to do. If you are trying to access a resource on the internet, such as a web page or an image, you should use a URL. A URL is a specific type of URI that specifies the location of a resource on the internet and includes a protocol at the beginning.

If you are trying to identify a resource by its name, rather than its location, you should use a URN (Uniform Resource Name). URNs are used to provide a persistent and location-independent way to identify resources.

If you are not sure which type of URI to use, you can use a generic URI. A URI is a generic term that refers to a string of characters that identify a resource. It can be used to identify any type of resource, not just those on the internet.
```

Reply:
some more comparisons for naming https://github.com/istio-ecosystem/authservice/blob/master/docs/README.md#message-config-configconfigproto

Reply:
using `url` sgtm !

Reply:
liking rfc in case others would be interested in further reading https://datatracker.ietf.org/doc/html/rfc2396#section-1.2

Reply:
Regarding `authorization_uri/url` and `token_uri/url` vs `authorization_endpoint` and `token_endpoint`, I think the former is more common in the wild, but the latter is more consistent with the OIDC spec. I'm fine with either but slightly prefer the latter because of consistency with the OIDC spec and Envoy's OAuth2 filter.

Reply:
ptal @envoyproxy/gateway-maintainers

Reply:
I think it's a good practice to use the same terms across all EG APIs if there're no existing well-known references to those terms. :-)

But my argument for using the same terms from the OIDC spec is that it's a well-known spec and it's easier for people to understand the terms if they are consistent with the spec. 

- If a user is familiar with OIDC, they might be confused if we use different terms in EG API.
- If a user is not familiar with OIDC, they may look up the OIDC spec to understand the terms, and they will also be confused if we use different terms in EG API.